PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-java-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/java/sdk/analyzer/DynamicClientCreationCheck.java

```diff
@@ -31,7 +32,7 @@ public class DynamicClientCreationCheck extends LocalInspectionTool {
     @NotNull
     @Override
     public JavaElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
-        return new DynamicClientCreationVisitor(holder, RuleConfigLoader.getInstance());
+        return new DynamicClientCreationVisitor(holder, RuleConfigLoader.getInstance().getRuleConfigs());
```

Comment:
I'm not quite sure, do you know if this could run prior to `ProjectActivity`? If that's the case, there might be a NPE, given that `INSTANCE` might not have initialized yet.

Reply:
I did do a runIde for testing this strategy. It does seem to be working as expected. I will try to look up some documentation related to this.